### PR TITLE
Diameter updates

### PR DESCRIPTION
**Pull request by younglululu**
fix lint warnings in DIAmeter

### CODE REVIEW

File: src/app/tide/spectrum_preprocess2.cc [hunk 1]
```diff
@@ -231,24 +184,20 @@ void ObservedPeakSet::PreprocessSpectrum(const Spectrum& spectrum, int charge,
...
+        for (int peak_idx=0; peak_idx<region_peaks.size(); ++peak_idx) {
+          if (peak_idx >= Params::GetInt("msamanda-regional-topk")) { break; }
+          // carp(CARP_DETAILED_DEBUG, "Region:[%d, %d] \t total_peaks: %d \t Peak mzbin: %d \t intensity: %f", i * region_size, (i+1) * region_size, region_peaks.size(), region_peaks[peak_idx].first, region_peaks[peak_idx].second );
+          // int peak_mzbin = region_peaks[peak_idx].first;
   	      dyn_filtered_peak_tuples_.push_back(region_peaks[peak_idx]);
+        }
       }
+
     }
     sort(dyn_filtered_peak_tuples_.begin(), dyn_filtered_peak_tuples_.end(), [](const pair<int, double> &left, const pair<int, double> &right) { return left.first < right.first; });
```
Comment (kfattila) [inline src/app/tide/spectrum_preprocess2.cc:200]: This sort should be in a dia_mode==true block too.

File: src/app/tide/spectrum_preprocess2.cc [hunk 2]
```diff
@@ -231,24 +184,20 @@ void ObservedPeakSet::PreprocessSpectrum(const Spectrum& spectrum, int charge,
...
 
       // added by Yang
-      // sort region_peaks w.r.t the descending intensity
-      sort(region_peaks.begin(), region_peaks.end(), [](const pair<int, double> &left, const pair<int, double> &right) { return left.second > right.second; });
-      // save the top samanda-regional-topk peaks per region
-      for (int peak_idx=0; peak_idx<region_peaks.size(); ++peak_idx) {
-    	  if (peak_idx >= Params::GetInt("msamanda-regional-topk")) { break; }
-    	  // carp(CARP_DETAILED_DEBUG, "Region:[%d, %d] \t total_peaks: %d \t Peak mzbin: %d \t intensity: %f", i * region_size, (i+1) * region_size, region_peaks.size(), region_peaks[peak_idx].first, region_peaks[peak_idx].second );
-    	  // int peak_mzbin = region_peaks[peak_idx].first;
+      if (dia_mode) {
```
Comment (kfattila) [inline src/app/tide/spectrum_preprocess2.cc:187]: The code line, above `region_peaks.push_back(make_pair(index, peaks_[index]));` should also be in a dia_mode block

**Review by kfattila — CHANGES_REQUESTED** [review]
There are several lines of codes in the src/app/tide/peptide.cpp file, in the void Peptide::AddIons(W* workspace)  fnc call. Those lines make the theoretical peptide generation quite slow, and those lines of code should be put in a "dia_mode" block too.

File: src/app/tide/spectrum_preprocess2.cc (hunk 1 above)
Comment (younglululu) [reply src/app/tide/spectrum_preprocess2.cc:200]: I was definitely aware of that. However, my thought was that the cost of sorting an empty vector is negligible, therefore I am inclined to make minimal changes by not adding dia_mode==true block. Could you let me know if you agree? Thanks!

**Comment by younglululu** [comment]
> There are several lines of codes in the src/app/tide/peptide.cpp file, in the void Peptide::AddIons(W* workspace) fnc call. Those lines make the theoretical peptide generation quite slow, and those lines of code should be put in a "dia_mode" block too.

Resolved. Please check the latest updates again. Thanks!

File: src/app/tide/spectrum_preprocess2.cc (hunk 2 above)
Comment (kfattila) [reply src/app/tide/spectrum_preprocess2.cc:187]: The issue in line 214 (or 167) is still there.

Comment (kfattila) [reply src/app/tide/spectrum_preprocess2.cc:187]: The issue in line 214 (or 167) is still there.

File: src/app/tide/spectrum_preprocess2.cc (hunk 1 above)
Comment (kfattila) [reply src/app/tide/spectrum_preprocess2.cc:199]: I think an 'if condition' would be faster, but I can settle down for your current approach.

**Comment by younglululu** [comment]
@kfattila Can I merge the pull request now?

**Review by kfattila — APPROVED** [review]
The code changes are OK with me, you may proceed merge these changes. Did you run the performance and the smoke tests, did they pass, did you test DIAMeter with the new code with your data?

**Comment by younglululu** [comment]
Hi Attila,

I did run the smoke tests and they passed. But I am not sure what are performance tests referring to.

**Comment by wsnoble** [comment]
The performance tests are in this directory:

https://github.com/crux-toolkit/crux-toolkit/tree/master/test/performance-tests

**Pull request merged**